codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -479,4 +274,153 @@ public HttpHeaders clear() {
         super.clear();
         return this;
     }
+
+    @Override
+    public int hashCode() {
+        return size();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof HttpHeaders)) {
+            return false;
+        }
+        HttpHeaders headers = (HttpHeaders) other;
+        if (size() != headers.size()) {
+            return false;
+        }
+        Comparator<CharSequence> valueComparator = AsciiString.CHARSEQUENCE_CASE_SENSITIVE_ORDER;
```

Comment:
Continuation of https://github.com/netty/netty/pull/3681#discussion_r35021121

me
> I think this code was shared before. Can we continue to share this (equality based upon a comparator)?

@buchgr
> @Scottmitch well not in `DefaultHeaders` as it doesn't know about `Comparators`.

That is fine. Doesn't have to be there, but I think we should still share code instead of duplicate.  Something like this:

```java
    public static <V> boolean equals(Headers<V> a, Headers<V> b, Comparator<V> valueComparator) {
        for (V name : a.names()) {
            List<V> otherValues = b.getAll(name);
            List<V> values = a.getAll(name);
            if (otherValues.size() != values.size()) {
                return false;
            }
            for (int i = 0; i < otherValues.size(); i++) {
                if (valueComparator.compare(otherValues.get(i), values.get(i)) != 0) {
                    return false;
                }
            }
        }
        return true;
    }
```